sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityToken.java

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.implementation;
+
+import com.azure.core.credential.AccessToken;
+
+import java.time.OffsetDateTime;
+
+/**
+ * Type representing authentication result from the azure-identity client.
+ */
+public class IdentityToken extends AccessToken {
+    /**
+     * Creates an identity token instance.
+     *
+     * @param token the token string.
+     * @param expiresAt the expiration time.
+     * @param options the identity client options.
+     */
+    public IdentityToken(String token, OffsetDateTime expiresAt, IdentityClientOptions options) {
+        super(token, expiresAt.plusMinutes(2).minus(options.getTokenRefreshOffset()));
```

Comment:
nit; `Object.requireNonNull` checks.

Reply:
discussed with Jinaghao offline, these are impl classes and these values are derived from service returned response. Hence we don't have to check here given we shouldn't test service behavior.